src/main/java/at/hannibal2/skyhanni/features/garden/visitor/GardenVisitorSupercraft.kt

```diff
@@ -0,0 +1,119 @@
+package at.hannibal2.skyhanni.features.garden.visitor
+
+import at.hannibal2.skyhanni.data.SackAPI
+import at.hannibal2.skyhanni.events.InventoryCloseEvent
+import at.hannibal2.skyhanni.events.garden.visitor.VisitorOpenEvent
+import at.hannibal2.skyhanni.test.command.ErrorManager
+import at.hannibal2.skyhanni.utils.ChatUtils
+import at.hannibal2.skyhanni.utils.ItemUtils
+import at.hannibal2.skyhanni.utils.ItemUtils.getLore
+import at.hannibal2.skyhanni.utils.ItemUtils.itemName
+import at.hannibal2.skyhanni.utils.NEUInternalName
+import at.hannibal2.skyhanni.utils.NEUInternalName.Companion.asInternalName
+import at.hannibal2.skyhanni.utils.NEUItems
+import at.hannibal2.skyhanni.utils.NEUItems.getItemStack
+import at.hannibal2.skyhanni.utils.SimpleTimeMark
+import io.github.moulberry.notenoughupdates.events.ReplaceItemEvent
+import io.github.moulberry.notenoughupdates.events.SlotClickEvent
+import io.github.moulberry.notenoughupdates.util.Utils
+import net.minecraft.entity.player.InventoryPlayer
+import net.minecraftforge.fml.common.eventhandler.EventPriority
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import kotlin.time.Duration.Companion.seconds
+
+class GardenVisitorSupercraft {
+
+    private val isSupercraftEnabled get() = VisitorAPI.config.shoppingList.showSuperCraft
+
+    private var hasIngredients = false;
+    private var lastClick = SimpleTimeMark.farPast()
+    private var lastSuperCraftMaterial = ""
+
+    private val superCraftItem by lazy {
+        val neuItem = "GOLD_PICKAXE".asInternalName().getItemStack()
+        Utils.createItemStack(
+            neuItem.item,
+            "§bSuper Craft",
+            "§7You have the items to craft",
+            "§7Click me to open the super crafter!"
+        )
+    }
+
+    @SubscribeEvent
+    fun onInventoryClose(event: InventoryCloseEvent) {
+        if (hasIngredients) {
+            hasIngredients = false
+        }
+    }
+
+    @SubscribeEvent(priority = EventPriority.HIGH)
+    fun onVisitorOpen(event: VisitorOpenEvent) {
+        val visitor = event.visitor
+        val offerItem = visitor.offer!!.offerItem
```

Comment:
Do not use `!!` in Kotlin ever, please.

If we don't have an offer item, we return here
e.g. 
```suggestion
        val offerItem = visitor.offer?.offerItem ?: return
```